python/pyspark/pandas/tests/test_categorical.py

```diff
@@ -243,18 +243,16 @@ def test_astype(self):
 
         self.assert_eq(kcser.astype("category"), pcser.astype("category"))
 
+        # CategoricalDtype is not updated if the dtype is same from pandas 1.3.
         if LooseVersion(pd.__version__) >= LooseVersion("1.3"):
-            # TODO(SPARK-36367): Fix the behavior to follow pandas >= 1.3
-            pass
-        elif LooseVersion(pd.__version__) >= LooseVersion("1.2"):
             self.assert_eq(
                 kcser.astype(CategoricalDtype(["b", "c", "a"])),
                 pcser.astype(CategoricalDtype(["b", "c", "a"])),
             )
         else:
             self.assert_eq(
                 kcser.astype(CategoricalDtype(["b", "c", "a"])),
-                pser.astype(CategoricalDtype(["b", "c", "a"])),
+                kcser,
```

Comment:
So when `LooseVersion(pd.__version__) < LooseVersion("1.3")`, `kcser` is not updated?

Reply:
btw, nit: this should be `pcser` instead of `kcser`?

Reply:
Yes, because `kcser` is pandas-on-Spark Series, which always follows latest pandas(1.3).

Reply:
Oh makes sense, thanks for the explanation.